pkgs/applications/video/lbry/default.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, fetchurl
+, stdenv
+, appimageTools
+, ffmpeg
+, gtk3
+, makeWrapper
+, xdg_utils
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lbry-desktop";
+  version = "0.50.2";
+
+  # Fetch and extract the AppImage
+  src = appimageTools.extract {
+    name = pname;
+    src = fetchurl {
+      url = "https://github.com/lbryio/lbry-desktop/releases/download/v${version}/LBRY_${version}.AppImage";
+      # Gotten from latest-linux.yml
+      sha512 = "br6HvVRz+ybmAhmQh3vOC5wgLmOCVrGHDn59ueWk6rFoKOCbm8WdmdadOZvHeN1ld2nlvPzEy+KXMOEfF1LeQg==";
+    };
+  };
+
+  buildInputs = [ ffmpeg gtk3 ];
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+
+    # place the extracted AppImage source in lib
+    LBRY_PATH="$out/lib/${pname}"
+    mkdir -p $LBRY_PATH
+    cp -r $src/* $LBRY_PATH
+
+    # executable wrapper
+    mkdir -p $out/bin
+    makeWrapper $LBRY_PATH/lbry $out/bin/lbry \
+      --prefix XDG_DATA_DIRS : $GSETTINGS_SCHEMAS_PATH \
+      --prefix PATH : ${ffmpeg}/bin:${xdg_utils}/bin
+
+    # desktop file/icon
+    mkdir -p $out/share/applications $out/share/icons/hicolor/256x256/apps
+    ln -s $LBRY_PATH/lbry.png $out/share/icons/hicolor/256x256/lbry.png
+    ln -s $LBRY_PATH/lbry.desktop $out/share/applications/lbry.desktop
+    substituteInPlace $LBRY_PATH/lbry.desktop \
+      --replace AppRun $out/bin/lbry
+
+    runHook postInstall
+  '';
+}
```

```suggestion
  meta = with lib; {
    homepage = "https://lbry.com/";
    description = "A graphical browser for the decentralized content marketplace provided by the LBRY protocol";
    license = licenses.mit;
    maintainers = with maintainers; [ enderger ];
    platforms = [ "x86_64-linux" ];
  };
}
```
You need a meta section, and you probably want to set yourself as maintainer.

---

Oh, oops! Sorry about that, I had a previous version trying to build from source but I couldn't get electron-builder to work and decided to switch to appimage (tried for ~3 days with no success)

---

Also, I can confirm that upstream does not support non-x86_64 (with ARM locked out due to a need to build from source using electron-builder, something which I have failed to do after much effort)